### PR TITLE
Fix #1393: FIDO2: externalize authenticator configuration

### LINKED ISSUE #1393: FIDO2: externalize authenticator configuration
Currently, Fido2Authenticators.java contains a hardcoded configuration for the FIDO authenticator.

Externalize the config so it can be changed by configuration. Suggestion DB - but be aware of the performance consequences.

The configuration changes are not expected to be often - in production for the first release and more frequent for setting in the test environments. So singleton with long eviction should be fine.

### PR DESCRIPTION
**Pull request by jnpsk**
To externalise authenticator configuration a new table `pa_fido2_authenticator` was created. It contains the same details about a fido2 authenticator as `Fido2Authenticators` did.   
The initial records are inserted to the table during liquibase update from `aaguid.csv` file converted from the `Fido2Authenticators` class.

To handle increased time complexity of db query, a spring caching mechanism is employed. The cache flush is scheduled task with fixed delay (defaults to 1hr).

This is just a draft of the initial approach. We could implement a REST API to manage the FIDO2 Authetnicators. Then the cache flushing could be handled on the table update. Does it worth it?

### CODE REVIEW

File: powerauth-fido2/src/main/java/com/wultra/powerauth/fido2/config/CachingConfig.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * @author Jan Pesek, jan.pesek@wultra.com
+ */
+@Configuration
+@Slf4j
+public class CachingConfig {
+
+    @CacheEvict(value = "fido2-authenticators-cache", allEntries = true)
+    @Scheduled(fixedDelayString = "${powerauth.service.scheduled.job.fido2AuthenticatorCacheEviction:3600000}")
+    public void evictFido2AuthenticatorCache() {
+        logger.info("Flush FIDO2 Authenticators cache.");
```
Comment (romanstrobl) [inline powerauth-fido2/src/main/java/com/wultra/powerauth/fido2/config/CachingConfig.java:38]: Can we switch the cache flush to `debug` level?

File: powerauth-fido2/src/main/java/com/wultra/powerauth/fido2/database/entity/Fido2AuthenticatorEntity.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import java.io.Serializable;
+
+/**
+ * Entity representing a FIDO2 Authenticator details.
+ *
+ * @author Jan Pesek, jan.pesek@wultra.com
+ */
+@Entity
+@Table(name = "pa_fido2_authenticator")
+@Data
```
Comment (romanstrobl) [inline powerauth-fido2/src/main/java/com/wultra/powerauth/fido2/database/entity/Fido2AuthenticatorEntity.java:35]: The `@Data` annotation produces `equals` and `hashCode` implementations which are not suitable for JPA classes. Can we add our own implementations?

**Review by romanstrobl — APPROVED** [review]
Good job. I found several minor issues.

File: powerauth-fido2/src/main/java/com/wultra/powerauth/fido2/database/entity/Fido2AuthenticatorEntity.java (hunk 2 above)
Comment (jnpsk) [reply powerauth-fido2/src/main/java/com/wultra/powerauth/fido2/database/entity/Fido2AuthenticatorEntity.java]: Thanks. Now both `Fido2Authenticator` and `Fido2AuthenticatorEntity` has `equals` and `hashCode` based on `aaguid` only.

File: docs/db/changelog/changesets/powerauth-java-server/1.7.x/db.changelog-version.xml [hunk 3]
```diff
@@ -6,5 +6,6 @@
     <include file="20240115-add-columns-fido2.xml" relativeToChangelogFile="true" />
     <include file="20240212-application-config.xml" relativeToChangelogFile="true" />
     <include file="20240222-add-tag-1.7.0.xml" relativeToChangelogFile="true" />
+    <include file="20240312-fido2-authenticator.xml" relativeToChangelogFile="true" />
```
Comment (romanstrobl) [inline docs/db/changelog/changesets/powerauth-java-server/1.7.x/db.changelog-version.xml:9]: The order seems incorrect, release tag should be after last change.

**Comment by romanstrobl** [comment]
We should update migration guide in `docs/PowerAuth-Server-1.7.0.md` to reflect DB changes.

**Comment by romanstrobl** [comment]
We should also update file `docs/sql/mssql/migration_1.6.0_1.7.0.sql`.

**Comment by jnpsk** [comment]
- Database new table was described in md files.
- New database structure png was generated.
- Sql scripts to manual migration were generated.

**Review by romanstrobl — APPROVED** [review]
Database documentation was updated.

File: docs/db/changelog/changesets/powerauth-java-server/1.7.x/20240312-fido2-authenticator.xml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  ~ GNU Affero General Public License for more details.
+  ~
+  ~ You should have received a copy of the GNU Affero General Public License
+  ~ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+  -->
+
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+                   https://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-latest.xsd">
```
Comment (banterCZ) [inline docs/db/changelog/changesets/powerauth-java-server/1.7.x/20240312-fido2-authenticator.xml:23]: 👀 Nice, but mind that in Maven, we have a transitive dependency on Liquibase 4.24.0 but in Docker, we still use 4.23.2 cc @zcgandcomp I do not expect any big discrepancy, just highlighting that.

File: docs/images/arch_db_structure.png
Comment (banterCZ) [inline docs/images/arch_db_structure.png:1]: We should agree on whether to use dark mode or not.

File: powerauth-fido2/src/main/java/com/wultra/powerauth/fido2/database/entity/Fido2AuthenticatorEntity.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    private String description;
+
+    @Enumerated(EnumType.STRING)
+    @Column(name = "signature_type", nullable = false)
+    private SignatureType signatureType;
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
```
Comment (banterCZ) [inline powerauth-fido2/src/main/java/com/wultra/powerauth/fido2/database/entity/Fido2AuthenticatorEntity.java:58]: Shouldn't we use `ProxyUtils`? https://github.com/wultra/development-internal/wiki/Equals-and-hashCode-in-JPA-entities

File: docs/images/arch_db_structure.png
Comment (jnpsk) [reply docs/images/arch_db_structure.png:1]: I used the white one to match developer portal colours. We used that one before.

File: powerauth-fido2/src/main/java/com/wultra/powerauth/fido2/database/entity/Fido2AuthenticatorEntity.java (hunk 5 above)
Comment (jnpsk) [reply powerauth-fido2/src/main/java/com/wultra/powerauth/fido2/database/entity/Fido2AuthenticatorEntity.java]: I haven't found any rules of when a Proxy object is created on top of an Entity. To be safe and consistent with our recommendations I added the `ProxyUtils`.

**Review by banterCZ — APPROVED** [review]
Looks good.

**Pull request merged**